src/test/java/unit/AuthStatus.java

```diff
@@ -39,6 +39,40 @@ void authStatusWhenLoggedIn() throws IOException {
         "auth status includes proxy group email",
         authStatus.proxyGroupEmail,
         CoreMatchers.not(emptyOrNullString()));
+    assertThat(
+        "auth status without workspace defined does not include pet SA email",
+        authStatus.serviceAccountEmail,
+        CoreMatchers.is(emptyOrNullString()));
+    assertTrue(authStatus.loggedIn, "auth status indicates user is logged in");
+  }
+
+  @Test
+  @DisplayName("auth status includes pet SA email when workspace is defined")
+  void authStatusWithCurrentWorkspace() throws IOException {
+    // select a test user and login
+    TestUsers testUser = TestUsers.chooseTestUserWithSpendAccess();
+    testUser.login();
+
+    // `terra workspace create`
+    TestCommand.runCommandExpectSuccess("workspace", "create");
+
+    // `terra auth status --format=json`
+    UFAuthStatus authStatus =
+        TestCommand.runAndParseCommandExpectSuccess(UFAuthStatus.class, "auth", "status");
+
+    // check that it says logged in and includes the user, proxy group, and pet SA emails
+    assertThat(
+        "auth status email matches test user",
+        authStatus.userEmail,
+        equalToIgnoringCase(testUser.email));
+    assertThat(
+        "auth status includes proxy group email",
+        authStatus.proxyGroupEmail,
+        CoreMatchers.not(emptyOrNullString()));
```

Comment:
you could use a pattern that matches emails if you want to be sure it's valid output.

Reply:
Good idea. Added.